scipy/linalg/flapack.pyf.src

```diff
@@ -1464,102 +1464,50 @@ interface
 
    end subroutine <prefix2c>gesvd_lwork
 
-   subroutine <prefix2>gels(trans,m,n,nrhs,a,lda,b,ldb,work,lwork,info)
+subroutine <prefix>gels(trans,m,n,nrhs,a,lda,b,ldb,work,lwork,info)
 
    ! lqr,x,info = gels(a,b,lwork=..,overwrite_a=False,overwrite_b=False)
    ! Solve Minimize 2-norm(A * X - B).
 
-     fortranname <prefix2>gels
      callstatement (*f2py_func)(trans,&m,&n,&nrhs,a,&lda,b,&ldb,work,&lwork,&info)
-     callprotoargument char*,int*,int*,int*,<ctype2>*,int*,<ctype2>*,int*,<ctype2>*,int*,int*
+     callprotoargument char*,int*,int*,int*,<ctype>*,int*,<ctype>*,int*,<ctype>*,int*,int*
 
-     character optional,intent(in),check(*trans=='N'||*trans=='T'):: trans = 'N'
-     integer depend(b),intent(hide):: nrhs = shape(b,1)
-     integer depend(a),intent(hide):: lda = shape(a,0)
+     character optional,intent(in),check(*trans=='N'||*trans==<'T',\0,'C',\2>):: trans = 'N'
      integer intent(hide),depend(a):: m = shape(a,0)
      integer intent(hide),depend(a):: n = shape(a,1)
-     integer depend(b),intent(hide):: ldb = shape(b,0)
-     integer optional,intent(in),depend(nrhs,m,n),&
-          check(lwork>=1||lwork==-1) &
-          :: lwork=MAX( MIN(m,n) + MAX(MIN(m,n), nrhs ), 1 )
-     <ftype2> dimension(m,nrhs),intent(in,out,copy,out=x) :: b
-     <ftype2> dimension(m,n),intent(in,out,copy,out=lqr) :: a
-     <ftype2> depend(lwork),dimension(MAX(lwork,1)),intent(hide,cache):: work
+     integer intent(hide),depend(b):: nrhs = shape(b,1)
+     <ftype> dimension(m,n),intent(in,out,copy,out=lqr):: a
+     integer intent(hide),depend(a):: lda = MAX(1,shape(a,0))
+     <ftype> intent(in,out,copy,out=x),depend(trans,m,n),dimension((*trans=='N'?m:n),nrhs),check(shape(b,0)==(*trans=='N'?m:n)) :: b
```

Comment:
what does `trans=='N'?m:n` do? I'm assuming though that we'll need some logic with `MAX(m,n)`, if you look at the gelss routines that's how they handle the b matrix

Reply:
Yes I probably overlooked the extended b vector part. The routine works when that part is fixed. I'll just sweep one more time for the corner cases and update.